app/services/file_manager/file_upload/file_upload.py

```diff
@@ -119,6 +119,57 @@ def assemble_path(
     return current_folder_node, parent_folder, create_folder_flag, target_folder
 
 
+def item_duplication_check(
+    create_folder_flag: bool, file_objects: List[FileObject], upload_client: UploadClient
+) -> List[FileObject]:
+    '''
+    Summary:
+        The function will check if the file is already uploaded on the platform.
+        If the file is already uploaded, the function will ask if to skip the file.
+        Otherwise, the function will exit
+    Parameter:
+        - create_folder_flag(bool): the flag to indicate if need to create new folder
+        - file_objects(List[FileObject]): the list of file object
+        - upload_client(UploadClient): the upload client object
+    Return:
+        - non_duplicate_file_objects(List[FileObject]): the list of file object that is not duplicated
+    '''
+
+    # make the file duplication check to allow folde merging
+    non_duplicate_file_objects = []
+    if create_folder_flag is True:
+        non_duplicate_file_objects = file_objects
+    else:
+        mhandler.SrvOutPutHandler.file_duplication_check()
+        duplicated_file = []
+        debug_logger.debug(f'upload batch size: {AppConfig.Env.upload_batch_size}')
+        for file_batchs in batch_generator(file_objects, batch_size=AppConfig.Env.upload_batch_size):
+            start_time = time.time()
+            non_duplicates, duplicate_path = upload_client.check_upload_duplication(file_batchs)
+            debug_logger.debug(f'Check duplication time: {time.time() - start_time:.2f}s')
+            non_duplicate_file_objects.extend(non_duplicates)
+            duplicated_file.extend(duplicate_path)
+
+        # if all file objects we check are already existed on the platform
+        # then we will exit the upload process
+        if len(non_duplicate_file_objects) == 0 and len(file_objects) != 0:
+            mhandler.SrvOutPutHandler.file_duplication_check_warning_with_all_same()
+            SrvErrorHandler.customized_handle(ECustomizedError.UPLOAD_CANCEL, if_exit=True)
```

Comment:
At this line, if we have no duplicated files why is the upload cancelled?

Reply:
oh yes the if statement check if `len(non_duplicate_file_objects) == 0` but users have specify some input file `len(file_objects) != 0`. That means all the input files from user are indeed already uploaded to backend.